### PR TITLE
v2

### PR DESCRIPTION
**Pull request by dvoet**

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "dosResolutionHost": "dcp.bionimbus.org",
+  "bondBaseUrl": ""
```
Comment (dvoet) [inline config.json]: needs to be real

File: index.js [hunk 2]
```diff
@@ -79,8 +80,54 @@ const handler = (req, res) => {
...
+                        res.status(502).send(err);
+                        return;
+                    };
+                    try {
+                        var parsedServiceAccountKey = JSON.parse(bondResponse.text);
+                    } catch(e) {
+                        res.status(500).send(`Service account key not in correct format`);
+                        return;
+                    };
+                    res.status(200).send({'dos': parsedData, 'credentials': parsedServiceAccountKey})
```
Comment (dvoet) [inline index.js]: intent is to return json that looks like
{ "dos": <full response from resolution>, "credentials": <full response from bond>}
I am not certain this does that

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+FROM node:6.14
+
+## Prepare the image
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update
+RUN apt-get install -y -qq --no-install-recommends unzip
+RUN apt-get clean
```
Comment (dvoet) [inline docker/Dockerfile]: put all the apt-get call into in RUN call: each RUN call results in a separate image so really the cleanup is meaningless on a separate line

same with below and wget, unzip, rm

**Comment by dvoet** [comment]
👍 (github won't let me approve my own PR)

File: config.json.ctmpl [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+    "dosResolutionHost": {{if eq (env "ENVIRONMENT") "prod"}}"dcp.bionimbus.org"{{else}}"dcp.bionimbus.org"{{end}},
+    "bondBaseUrl": {{if eq (env "ENVIRONMENT") "prod"}}"broad-bond-prod.appspot.com"{{else if eq (env "ENVIRONMENT") "staging"}}"broad-bond-staging.appspot.com"{{else if eq (env "ENVIRONMENT") "alpha"}}"broad-bond-staging.appspot.com"{{else}}"broad-bond-dev.appspot.com"{{end}}
```
Comment (MatthewBemis) [inline config.json.ctmpl]: if you put `{{with $environment := env "ENVIRONMENT"}}` at the top of the file and `{{end}}` at the bottom you can simplify this line to:

`"bondBaseUrl": "broad-bond-{{$environment}}.appspot.com"`

File: index.js [hunk 5]
```diff
@@ -79,8 +80,53 @@ const handler = (req, res) => {
...
+                        res.status(502).send(bondErr);
+                        return;
+                    }
+                    try {
+                        var parsedServiceAccountKey = JSON.parse(bondResponse.text);
+                    } catch(e) {
+                        res.status(500).send(`Service account key not in correct format`);
+                        return;
+                    }
+                    res.status(200).send({'dos': parsedData, 'credentials': parsedServiceAccountKey["data"]});
```
Comment (dvoet) [inline index.js]: I have changed my mind - we should call credentials googleServiceAccount... I am thinking of the day where we have different types of credentials and it seems easier to name them differently than have a generic "credentials" entry with maybe different types inside.

**Pull request merged**